server/games/game.py

```diff
@@ -132,12 +133,15 @@ def __init__(
         self.game_options.add_callback("Title", self.on_title_changed)
 
         self.mods = {}
-        self._hosted_future = asyncio.Future()
         self._finish_lock = asyncio.Lock()
 
         self._logger.debug("%s created", self)
         asyncio.get_event_loop().create_task(self.timeout_game(setup_timeout))
 
+    @cached_property
+    def _hosted_future(self) -> asyncio.Future:
+        return asyncio.get_event_loop().create_future()
```

Comment:
Should we be using `get_running_loop` instead maybe? I'm not entirely sure what all the intricacies are but the asyncio docs for `get_event_loop` in python 3.10 say this:

> Because this function has rather complex behavior (especially when custom event loop policies are in use), using the [get_running_loop()](https://docs.python.org/3.10/library/asyncio-eventloop.html?highlight=get_event_loop#asyncio.get_running_loop) function is preferred to [get_event_loop()](https://docs.python.org/3.10/library/asyncio-eventloop.html?highlight=get_event_loop#asyncio.get_event_loop) in coroutines and callbacks.